server/controllers/booking/booking.controller.js

```diff
@@ -43,13 +50,66 @@ const createBooking = function(req, res) {
             console.log(validateErrs);
             res.status(500).send(errs);
           } else {
-            newBooking.save((err, booking) => {
-              if (err) {
-                res.status(504).send(err);
-              } else {
-                res.json(booking._id);
-              }
-            });
+            Booking.find({ car: req.body.car, disabled: false })
+              // Placeholder, should look for booking belonging to current user
+              .exec((err, bookings) => {
+                if (err) {
+                  res.status(500).send(err);
+                } else {
+                  // found match for car, start comparing time
+                  if (!bookings) {
+                    // No matching bookings -> ok to book
+                    newBooking.save((err, booking) => {
+                      if (err) {
+                        res.status(500).send(err);
+                      } else {
+                        res.status(200).send(booking._id);
+                      }
+                    });
+                  } else {
+                    //checking of time to be implemented
+                    bookings.forEach(b => {
+                      var compStart = moment(bookings.startsAt);
```

Comment:
This won't work as you define b in forEach but use bookings to reference it

Reply:
thank you 😀